docs/content/docs/recipes/collaborative-text-area.md

```diff
@@ -0,0 +1,325 @@
+        const charactersDeleted = oldText.length - newText.length;
+        sharedStringHelper.removeText(newCaretPosition, newCaretPosition + charactersDeleted);
+      }
+    };
+    ```
+
+1. Replace `TODO 3` with the following code. This function sets the selection directly in the `textarea` element.
+
+    ```js
+    const setTextareaSelection = (newStart, newEnd) => {
+      if (!textareaRef.current) {
+        throw new Error("Trying to set selection without current textarea ref?");
+      }
+      const textareaElement = textareaRef.current;
+      textareaElement.selectionStart = newStart;
+      textareaElement.selectionEnd = newEnd;
+    };
+    ```
+
+1. Replace `TODO 4` with the following code. This function sets the selection from the `textarea` element and sets it in the React refs.
+
+    ```js
+    const storeSelectionInReact = () => {
+      if (!textareaRef.current) {
+        throw new Error("Trying to remember selection without current textarea ref?");
+      }
+      const textareaElement = textareaRef.current;
+
+      const textareaSelectionStart = textareaElement.selectionStart;
+      const textareaSelectionEnd = textareaElement.selectionEnd;
+      selectionStartRef.current = textareaSelectionStart;
+      selectionEndRef.current = textareaSelectionEnd;
+    };
+    ```
+
+1. Replace `TODO 5` with the following code. Note about this code:
+    - By setting the dependency array at the end of `useEffect` to include `sharedStringHelper`, it is ensured that this function is called each time the `sharedStringHelper` object is changed.
+
+    ```js
+    React.useEffect(() => {
+      const handleTextChanged = (event) => {
+        const newText = sharedStringHelper.getText();
+        setText(newText);
+        if (!event.isLocal) {
+          const newSelectionStart = event.transformPosition(selectionStartRef.current);
+          const newSelectionEnd = event.transformPosition(selectionEndRef.current);
+          setTextareaSelection(newSelectionStart, newSelectionEnd);
+          storeSelectionInReact();
+        }
+      };
+
+      sharedStringHelper.on("textChanged", handleTextChanged);
+      return () => {
+        sharedStringHelper.off("textChanged", handleTextChanged);
+      };
+    }, [sharedStringHelper]);
+    ```
+
+1. Finally, replace `TODO 6` with the following code to create the `textarea` element.
+
+    ```js
+    return (
+    <textarea
+      rows={20}
+        cols={50}
+        ref={textareaRef}
+        onBeforeInput={storeSelectionInReact}
+        onKeyDown={storeSelectionInReact}
+        onClick={storeSelectionInReact}
+        onContextMenu={storeSelectionInReact}
+        onChange={handleChange}
+        value={text} />
+    );
+    ```
+
+## Start the Fluid server and run the application
+
+In the Command Prompt, run the following command to start the Fluid service. Note that `tinylicious` is the name of the Fluid service that runs on localhost.
+
+```dotnetcli
+npx tinylicious
```

Comment:
do we need the instruction to install this first?

Reply:
In this case, since we started from `create-react-app` we don't need to install. However, I just realized that I need to update more of this because of the switch to TypeScript.